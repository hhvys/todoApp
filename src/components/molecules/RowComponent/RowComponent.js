import React from 'react';
import styles from './RowComponent.mod.scss';
import Icon from "../../atoms/Icons/Icon";

class RowComponent extends React.Component {

	handleHeaderIconClick = () => {
		const {onHeaderIconClick, headerClickArgs} = this.props;
		if(onHeaderIconClick && headerClickArgs)
			onHeaderIconClick(...headerClickArgs);
	};

	handleFooterIconClick = () => {
		const {onFooterIconClick, footerClickArgs} = this.props;
		if(onFooterIconClick && footerClickArgs)
			onFooterIconClick(...footerClickArgs);
	};

	renderFooter = () => {
		const {
			footerIcon,
			footerClass
		} = this.props;

		return footerIcon ? (
			<div className={`full-height d-flex align-items-center justify-content-center`}>
				<Icon className={`${footerClass ? footerClass : ''} ${styles.endIcon}`}
							onClick={this.handleFooterIconClick}
							iconType={footerIcon}
				/>
			</div>
		) : null;
	};

	render() {
		const {
			className,
			headerClass,
			onClick,
			headerIcon,
			footerContent,
			mainContent,
			extraContent,
		} = this.props;

		return (
			<div
				className={`${className ? className : ''} ${styles.verticalTab} full-width d-flex align-items-center justify-content-center`}
				onClick={onClick}
			>

				{/*headerDiv*/}
				<div className={`${styles.verticalTab} full-height d-flex align-items-center justify-content-center`}>
					<Icon className={`${headerClass ? headerClass : ''} `} onClick={this.handleHeaderIconClick}
								iconType={headerIcon}/>
				</div>

				{/*contentDiv*/}
				<div className={`${styles.content} d-flex flex-column justify-content-center `}>
					<div className={`txt-overflow ml-2`}>
						{mainContent}
					</div>

					{
						extraContent ?
							(
								<div className={`${styles.extraContent} d-flex align-items-center ml-2`}>
									{extraContent}
								</div>
							) :
							null
					}

				</div>

				{/*footerDiv*/}
				<div className={`${styles.footer} full-height d-flex align-items-center justify-content-between`}>

					<div className={`${styles.footerContent} full-height d-flex align-items-center justify-content-end`}>
						{footerContent}
					</div>

					{this.renderFooter()}

				</div>

			</div>
		)
	}
}

export default RowComponent;
