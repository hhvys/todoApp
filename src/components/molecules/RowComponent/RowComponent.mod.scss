.verticalTab {
    background-color: inherit;
    cursor: pointer;
}

.header {
    width: 30px;
}

.content {
    flex: 1;
    height: 100%;
    width: 70%;
}

.extraContent {
    font-size: 10px;
    color: #262626;
}

.footer {
    width: 40px;
}

.footerContent {
    font-size: 12px;
    font-weight: 500;
    color: #bdbcbb;
    width: 28px;
}

.endIcon:hover {
    font-weight: bold;
}
