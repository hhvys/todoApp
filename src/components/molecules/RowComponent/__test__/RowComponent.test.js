import {shallow, configure} from 'enzyme';
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import RowComponent from '../RowComponent';

configure({adapter: new Adapter()});

// describe('RowComponent', () => {
//
// 	it('should pass className prop to wrapper div', () => {
// 		props.className = "randomClass";
// 		const divs = inputWithLabel().find('div');
// 		const wrapperDiv = divs.first();
// 		expect(wrapperDiv.props().className).toMatch(props.className);
// 	});
//
//
//
// });
