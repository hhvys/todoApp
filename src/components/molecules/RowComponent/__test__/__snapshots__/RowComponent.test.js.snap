// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RowComponent footerDiv should pass footerContent to footerDiv 1`] = `
<div
  className="footer full-height d-flex align-items-center justify-content-between"
>
  <div
    className="footerContent full-height d-flex align-items-center justify-content-end"
  >
    footerContent
  </div>
</div>
`;

exports[`RowComponent footerDiv should render footerIconDiv if footerSymbol is defined 1`] = `
<div
  className="footer full-height d-flex align-items-center justify-content-between"
>
  <div
    className="footerContent full-height d-flex align-items-center justify-content-end"
  >
    footerContent
  </div>
  <div
    className="full-height d-flex align-items-center justify-content-center"
  >
    <Icon
      className="footerClass endSymbol"
      height={20}
      onClick={[Function]}
      iconType="footerLogo"
      width={20}
    />
  </div>
</div>
`;

exports[`RowComponent headerDiv should pass header props to headerDiv 1`] = `
<div
  className="verticalTab full-height d-flex align-items-center justify-content-center"
>
  <Icon
    className="headerClass "
    height={20}
    onClick={[Function]}
    iconType="headerLogo"
    width={20}
  />
</div>
`;

exports[`RowComponent mainDiv should pass mainContent to contentDiv 1`] = `
<div
  className="content d-flex flex-column justify-content-center "
>
  <div
    className="d-flex align-items-center txt-overflow ml-2"
  >
    mainContent
  </div>
</div>
`;

exports[`RowComponent mainDiv should render extraContent div if extraContent is defined 1`] = `
<div
  className="content d-flex flex-column justify-content-center "
>
  <div
    className="d-flex align-items-center txt-overflow ml-2"
  >
    mainContent
  </div>
  <div
    className="extraContent d-flex align-items-center ml-2"
  >
    extraContent
  </div>
</div>
`;

exports[`RowComponent should render 1`] = `
<div
  className=" verticalTab full-width d-flex align-items-center justify-content-center "
>
  <div
    className="verticalTab full-height d-flex align-items-center justify-content-center"
  >
    <Icon
      className=" "
      height={20}
      width={20}
    />
  </div>
  <div
    className="content d-flex flex-column justify-content-center "
  >
    <div
      className="d-flex align-items-center txt-overflow ml-2"
    />
  </div>
  <div
    className="footer full-height d-flex align-items-center justify-content-between"
  >
    <div
      className="footerContent full-height d-flex align-items-center justify-content-end"
    />
  </div>
</div>
`;
